src/back/msl/writer.rs

```diff
@@ -815,8 +815,21 @@ impl<W: Write> Writer<W> {
                     crate::BinaryOperator::ShiftLeft => "<<",
                     crate::BinaryOperator::ShiftRight => ">>",
                 };
-                let kind = context
-                    .resolve_type(left)
+
+                let left_type = context.resolve_type(left);
+                let right_type = context.resolve_type(right);
+
+                let is_matrix_mul_tri_vec =
+                    op == crate::BinaryOperator::Multiply &&
+                    matches!(left_type, &crate::TypeInner::Matrix { ..}) &&
```

Comment:
can we please not use `matches!`? a regular match would take the same amount of code, and involve less sugar